src/gui/layout/qgslayoutappmenuprovider.h

```diff
@@ -16,28 +16,29 @@
 #ifndef QGSLAYOUTAPPMENUPROVIDER_H
 #define QGSLAYOUTAPPMENUPROVIDER_H
 
+#include "qgis_gui.h"
 #include "qgis_sip.h"
 #include "qgslayoutview.h"
 #include <QObject>
 
-class QgsLayoutDesignerDialog;
+class QgsLayoutDesignerInterface;
 
 /**
  * A menu provider for QgsLayoutView
  */
-class QgsLayoutAppMenuProvider : public QObject, public QgsLayoutViewMenuProvider
+class GUI_EXPORT QgsLayoutAppMenuProvider : public QObject, public QgsLayoutViewMenuProvider
```

Comment:
Instead of moving this one (which should be strongly tied to the designer dialog), couldn't you use the QgsLayoutViewMenuProvider interface instead to implement your own provider?

Reply:
Sure, no problem copying it to the KADAS tree. Actually though there is the `QgsLayoutDesignerInterface` which looked like an attempt to decouple this and other classes from the actual designer implementation. Looked like a good way to spare some redundant copy-pasted code?

Reply:
With QgsLayoutDesignerInterface I'm taking a super-conservative approach and only exposing a minimal set of methods in it (my next comment was going to be to see if you could avoid the newly added members in it :smile:) I''d be happier if we could avoid this particular move if you're ok with that.

Reply:
Ok - so I take I should also revert the `qgsreport*` classes switching from taking a `QgsLayoutDesignerDialog` to a `QgsLayoutDesignerInterface`?

Reply:
yes please!

Reply:
Ok should look a bit less scary now.